generator/src/main/java/org/corfudb/generator/operations/SnapshotTxOperation.java

```diff
@@ -12,31 +17,55 @@
 @Slf4j
 public class SnapshotTxOperation extends Operation {
 
+
     public SnapshotTxOperation(State state) {
         super(state);
+        shortName = "TxSnap";
     }
 
     @Override
     public void execute() {
-        long trimMark = state.getTrimMark();
-        Random rand = new Random();
-        long delta = (long) rand.nextInt(10) + 1;
-        state.startSnapshotTx(trimMark + delta);
-
-        int numOperations = state.getOperationCount().sample(1).get(0);
-        List<Operation> operations = state.getOperations().sample(numOperations);
-
-        for (int x = 0; x < operations.size(); x++) {
-            if (operations.get(x) instanceof OptimisticTxOperation
-                    || operations.get(x) instanceof SnapshotTxOperation
-                    || operations.get(x) instanceof RemoveOperation
-                    || operations.get(x) instanceof WriteOperation) {
-                continue;
+        try {
+            long trimMark = state.getTrimMark();
+            Random rand = new Random();
+            long delta = (long) rand.nextInt(10) + 1;
+
+            // Safety Hack for not having snapshot in the future
+
+            long current_max = state.getRuntime().getSequencerView()
+                    .nextToken(Collections.emptySet(), 0)
+                    .getToken().getTokenValue();
+
+            long snapShotAddress = Long.min(trimMark + delta, current_max);
+
+
+            String txStartRecord = String.format("%s, %s", shortName, "start");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "%s, %s" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)